### PR TITLE
HPCC-30327 ECL Watch v9 spray multiple files open multiple tabs to WUs

### PR DESCRIPTION
**Pull request by jeclrsg**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx [hunk 1]
```diff
@@ -88,23 +90,42 @@ export const BlobImportForm: React.FunctionComponent<BlobImportFormProps> = ({
                     }
                     request["sourcePath"] = file.SourceFile;
                     request["fullPath"] = file.SourceFile;
-                    FileSpray.SprayFixed({
+                    requests.push(FileSpray.SprayFixed({
                         request: request
-                    }).then((response) => {
+                    }));
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/BlobImportForm.tsx:95]: Error reporting

File: esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx [hunk 2]
```diff
@@ -109,23 +111,42 @@ export const DelimitedImportForm: React.FunctionComponent<DelimitedImportFormPro
                         file.TargetName && file.TargetName.substring(0, 2) !== "::"
                     ) ? "::" : "") + file.TargetName;
                     request["destNumParts"] = file.NumParts;
-                    FileSpray.SprayVariable({
+                    requests.push(FileSpray.SprayVariable({
                         request: request
-                    }).then((response) => {
+                    }));
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/DelimitedImportForm.tsx:116]: Error reporting

File: esp/src/src-react/components/forms/landing-zone/FixedImportForm.tsx [hunk 3]
```diff
@@ -93,23 +95,42 @@ export const FixedImportForm: React.FunctionComponent<FixedImportFormProps> = ({
                         file.TargetName && file.TargetName.substring(0, 2) !== "::"
                     ) ? "::" : "") + file.TargetName;
                     request["destNumParts"] = file.NumParts;
-                    FileSpray.SprayFixed({
+                    requests.push(FileSpray.SprayFixed({
                         request: request
-                    }).then((response) => {
+                    }));
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/FixedImportForm.tsx:100]: Error reporting

File: esp/src/src-react/components/forms/landing-zone/JsonImportForm.tsx [hunk 4]
```diff
@@ -97,23 +99,42 @@ export const JsonImportForm: React.FunctionComponent<JsonImportFormProps> = ({
                     ) ? "::" : "") + file.TargetName;
                     request["sourceRowTag"] = file.TargetRowPath;
                     request["destNumParts"] = file.NumParts;
-                    FileSpray.SprayVariable({
+                    requests.push(FileSpray.SprayVariable({
                         request: request
-                    }).then((response) => {
+                    }));
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/JsonImportForm.tsx:104]: Error reporting

File: esp/src/src-react/components/forms/landing-zone/VariableImportForm.tsx [hunk 5]
```diff
@@ -87,26 +89,45 @@ export const VariableImportForm: React.FunctionComponent<VariableImportFormProps
...
-                        data.namePrefix && data.namePrefix.substr(-2) !== "::" &&
-                        file.TargetName && file.TargetName.substr(0, 2) !== "::"
+                        data.namePrefix && data.namePrefix.substring(-2) !== "::" &&
+                        file.TargetName && file.TargetName.substring(0, 2) !== "::"
                     ) ? "::" : "") + file.TargetName;
-                    FileSpray.SprayFixed({
+                    requests.push(FileSpray.SprayFixed({
                         request: request
-                    }).then((response) => {
+                    }));
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/VariableImportForm.tsx:97]: Error reporting

File: esp/src/src-react/components/forms/landing-zone/XmlImportForm.tsx [hunk 6]
```diff
@@ -96,23 +98,42 @@ export const XmlImportForm: React.FunctionComponent<XmlImportFormProps> = ({
                     ) ? "::" : "") + file.TargetName;
                     request["sourceRowTag"] = file.TargetRowTag;
                     request["destNumParts"] = file.NumParts;
-                    FileSpray.SprayVariable({
+                    requests.push(FileSpray.SprayVariable({
                         request: request
-                    }).then((response) => {
+                    }));
```
Comment (GordonSmith) [inline esp/src/src-react/components/forms/landing-zone/XmlImportForm.tsx:103]: Error reporting

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/forms/landing-zone/XmlImportForm.tsx (hunk 6 above)
Comment (jeclrsg) [reply esp/src/src-react/components/forms/landing-zone/XmlImportForm.tsx:103]: Added a .catch(...) a few lines below this comment, after the `Promise.all(requests).then(...)` in each of these forms.

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**